subsys/mgmt/mcumgr/smp_pkasm.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <net/buf.h>
+#include <net/net_ip.h>
+#include <mgmt/mcumgr/buf.h>
+#include <mgmt/mcumgr/smp.h>
+#include <mgmt/mgmt.h>
+#include <smp/smp.h>
+
+int zephyr_smp_pkasm_collect(struct zephyr_smp_transport *zst, const void *buf, uint16_t len)
+{
+	/* No allocated buffer indicates that we are expecting first fragment */
+	if (zst->_pkasm.current == NULL) {
+		if (len >= sizeof(struct mgmt_hdr)) {
+			uint16_t expected = ntohs(((struct mgmt_hdr *)buf)->nh_len);
```

Comment:
Can we change this to general converter API: _sys_be16_to_cpu_ and get rid of the _net_ip_ header? This won't necessarily have to process IP packets so the chosen header may be misleading.

Reply:
Done